pandas/core/strings.py

```diff
@@ -2246,21 +2239,21 @@ def cat(self, others=None, sep=None, na_rep=None, join=None):
                           "'outer'|'inner'|'right'`. The future default will "
                           "be `join='left'`.", FutureWarning, stacklevel=2)
 
-        # concatenate others into DataFrame; need to add keys for uniqueness in
-        # case of duplicate columns (for join is None, all indexes are already
-        # the same after _get_series_list, which forces alignment in this case)
-        others = concat(others, axis=1,
```

Comment:
@WillAyd, this is the main reason for the slow-down. For working with a `DataFrame` below (as you wished), we first need to create it with `pd.concat` (expensive). Before, we were only using `pd.concat` if the indices need to be aligned (which they don't in the benchmarks).